spec.html

```diff
@@ -32006,7 +32006,7 @@
           1. Let _O_ be the *this* value.
           1. Perform ? ValidateTypedArray(_O_).
           1. Let _value_ be ? ToNumber(_value_).
-          1. Let _len_ be ? _O_.[[ArrayLength]].
+          1. Let _len_ be _O_.[[ArrayLength]].
```

Comment:
while this fix is ok, it feels like we miss a point between the previous and next steps to throw in case the TypedArray has a detachedbuffer.

It makes me wonder if we should prepend this step with something like:

```
If IsDetachedBuffer(O.[[ViewedArrayBuffer]]) is true, throw a TypeError exception.
```

Of just move this `Let len be _O_.[[ArrayLength]]` to run before `Let _value_ be ? ToNumber(_value_)`